includes/events/class-omise-event-charge-create.php

```diff
@@ -1,88 +1,68 @@
-	 * payment methods that would trigger the 'charge.create' event.
-	 *
-	 * =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-	 * Alipay
-	 * charge data in payload:
-	 *     [status: 'pending' (always)], [authorized: 'false' (always)]
-	 *
-	 * =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-	 * Internet Banking
-	 * charge data in payload:
-	 *     [status: 'pending' (always)], [authorized: 'false' (always)]
-	 *
-	 * =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-	 * Credit Card (none 3-D Secure)
-	 * CAPTURE = FALSE
-	 * charge data in payload could be one of these sets:
-	 *     [status: 'pending'], [authorized: 'true'], [paid: 'false']
-	 *     [status: 'failed'], [authorized: 'false'], [paid: 'false']
-	 *
-	 * CAPTURE = TRUE
-	 * charge data in payload could be one of these sets:
-	 *     [status: 'successful'], [authorized: 'true'], [paid: 'true']
-	 *     [status: 'failed'], [authorized: 'false'], [paid: 'false']
-	 *
-	 * =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
-	 * Credit Card (3-D Secure)
-	 * CAPTURE = FALSE
-	 * charge data in payload could be one of these sets:
-	 *     [status: 'pending'], [authorized: 'false'], [paid: 'false']
-	 *     [status: 'failed'], [authorized: 'false'], [paid: 'false']
-	 *
-	 * CAPTURE = TRUE
-	 * charge data in payload could be one of these sets:
-	 *     [status: 'pending'], [authorized: 'false'], [paid: 'false']
-	 *     [status: 'failed'], [authorized: 'false'], [paid: 'false']
-	 *
-	 * @param  mixed $data
-	 *
-	 * @return void
+	 * @inheritdoc
 	 */
-	public function handle( $data ) {
-		if ( 'charge' !== $data->object || ! isset( $data->metadata->order_id ) ) {
-			return;
+	public function validate() {
+		if ( 'charge' !== $this->data['object'] || ! isset( $this->data['metadata']['order_id'] ) ) {
+			return false;
 		}
 
-		if ( ! $order = wc_get_order( $data->metadata->order_id ) ) {
-			return;
+		if ( ! $this->order = wc_get_order( $this->data['metadata']['order_id'] ) ) {
+			return false;
 		}
 
-		$order->add_order_note(
-			__(
-				'Omise: an event charge.create has been caught (webhook).',
-				'omise'
-			)
-		);
-
-		/** 
-		 * Note. There is no special case for 'charge.create' event to handle with.
-		 *       Basically, just to pass this event in case some 3rd-party developer
-		 *       need to add extra process, then they can hook 'omise_handled_event_charge_create' action.
-		 */
-
-		/**
-		 * Hook after Omise handle an event from webhook.
-		 *
-		 * @param WC_Order $order  an order object.
-		 * @param mixed $data      a data of an event object
-		 */
-		do_action( 'omise_handled_event_charge_create', $order, $data );
+		// Making sure that an event's charge id is identical with an order transaction id.
+		if ( $this->order->get_transaction_id() !== $this->data['id'] ) {
+			return false;
+		}
 
-		return;
+		return true;
 	}
 }
```

Comment:
Removing all the code here as this event does not need any extra step/action from Webhook (however, still keep this handler class here to support those 3rd-party developers that may need to hook this Webhook event).